pandas/core/indexes/datetimelike.py

```diff
@@ -855,3 +753,70 @@ def f(self):
     f.__name__ = fget.__name__
     f.__doc__ = fget.__doc__
     return property(f)
+
+
+class DatetimelikeDelegateMixin(PandasDelegate):
```

Comment:
This refactor, and the changes in PeriodDelegateMixin have been split off into #24039.

The new classes DatetimeDelegateMixin (`indexes/datetimes.py`) and `TimedeltaDelegateMixin` (`indexes/timedelta.py`) will be implemented here, and should give some motivation for #24039.